CMakeLists.txt

```diff
@@ -531,23 +540,12 @@ endif()
 # Give access to functions defined in internal.h when BUILD_SHARED_LIBS is ON, to tests for example.
 # The avif_internal target should not be used by external code.
 if(BUILD_SHARED_LIBS)
-    add_library(avif_internal STATIC "$<TARGET_PROPERTY:avif,SOURCES>")
-    # Copy or duplicate most properties from the public avif library target.
-    target_compile_definitions(avif_internal PRIVATE "$<TARGET_PROPERTY:avif,COMPILE_DEFINITIONS>")
+    add_library(avif_internal STATIC)
+    target_link_libraries(avif_internal PRIVATE avif_obj)
     target_include_directories(avif_internal PUBLIC ${libavif_SOURCE_DIR}/include)
-    get_property(AVIF_INTERNAL_LINK_LIBRARIES TARGET avif PROPERTY LINK_LIBRARIES)
-    target_link_libraries(avif_internal PRIVATE ${AVIF_INTERNAL_LINK_LIBRARIES})
-    get_property(AVIF_INTERNAL_COMPILE_OPTIONS TARGET avif PROPERTY COMPILE_OPTIONS)
-    target_compile_options(avif_internal PUBLIC ${AVIF_INTERNAL_COMPILE_OPTIONS})
-    if(AVIF_USE_CXX)
-        set_target_properties(avif_internal PROPERTIES LINKER_LANGUAGE "CXX")
-    endif()
-    if(NOT AVIF_LIBYUV_ENABLED)
-        target_include_directories(avif_internal PRIVATE ${libavif_SOURCE_DIR}/third_party/libyuv/include/)
-    endif()
-    if(AVIF_ENABLE_COMPLIANCE_WARDEN)
-        target_include_directories(avif_internal PRIVATE ${libavif_SOURCE_DIR}/ext/ComplianceWarden/src/utils/)
-    endif()
+    # Define the following to avoid linking against the two at the same time.
+    target_compile_definitions(avif PUBLIC $<BUILD_INTERFACE:AVIF_DLL>)
```

Comment:
This seems redundant with line 525 above. Can we delete this?

If we delete this line, we need to update the comment at line 546.